src/sys.cpp

```diff
@@ -0,0 +1,26 @@
+#include "sys.h"
+
+#include "stdint.h"
+
+void syscall_handler(void* syscall_frame) {
```

Comment:
- would be good to parse arguments already so implementation is easier for others.
- also, i wonder how this design works w/ syscalls being called from code that matches the linux ABI. we wanted to support that right?